File: app/src/main/java/com/roman/gurdan/sudo/pro/view/DataView.java
package com.roman.gurdan.sudo.pro.view;

import android.content.Context;
import android.content.res.TypedArray;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.cardview.widget.CardView;

import com.roman.gurdan.sudo.pro.R;

public class DataView extends CardView {

    private TextView title;
    private TextView msg;

    private int titleRes;
    private int msgRes;

    public DataView(@NonNull Context context) {
        this(context, null);
    }

    public DataView(@NonNull Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, -1);
    }

    public DataView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);

        TypedArray array = context.obtainStyledAttributes(attrs, R.styleable.DataView);
        titleRes = array.getResourceId(R.styleable.DataView_vTitle, R.string.nan);
        msgRes = array.getColor(R.styleable.DataView_vMsgColor, context.getResources().getColor(R.color.teal_200));
        array.recycle();
    }

    @Override
    protected void onAttachedToWindow() {
        super.onAttachedToWindow();
        try {
            View.inflate(getContext(),R.layout.view_data_item, this);
            title = findViewById(R.id.title);
            msg = findViewById(R.id.msg);
            title.setText(titleRes);
            msg.setTextColor(msgRes);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public void setMsg(String data) {
        if (this.msg != null && data != null)
            this.msg.setText(data);
    }


}

File: app/src/main/java/com/roman/gurdan/sudo/pro/view/DataView.kt
package com.roman.gurdan.sudo.pro.view

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.TextView
import androidx.cardview.widget.CardView
import com.roman.gurdan.sudo.pro.R

class DataView(context: Context, attrs: AttributeSet?) :
    CardView(context, attrs) {

    private lateinit var title: TextView
    private lateinit var msg: TextView
    private var titleRes: Int = 0
    private var msgRes: Int = 0

    init {
        context.obtainStyledAttributes(attrs, R.styleable.DataView).apply {
            titleRes = this.getResourceId(R.styleable.DataView_vTitle, R.string.nan)
            msgRes = this.getColor(
                R.styleable.DataView_vMsgColor,
                context.resources.getColor(R.color.teal_200)
            )
            this.recycle()
        }
    }

    override fun onAttachedToWindow() {
        super.onAttachedToWindow()
        try {
            View.inflate(context, R.layout.view_data_item, this)
            title = findViewById(R.id.title)
            msg = findViewById(R.id.msg)

            title.setText(titleRes)
            msg.setTextColor(msgRes)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun setMsg(data: String) {
        try {
            this.msg.text = data
        } catch (e: Exception) {
        }
    }


}